### PR TITLE
[ie/reddit] Improve HTTP 403 error message

### LINKED ISSUE #10924: [Reddit] Failed to parse JSON
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

I tried to download a link from reddit  which is a normal video and I got an error message.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
ubuntu@instance-20240830-1406:~$ yt-dlp -vU https://www.reddit.com/r/funnyvideos/comments/1f8xxze/three_hairy_guys_doing_some_practicals/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
[1] 2190997
[2] 2190998
[3] 2190999
[4] 2191000
[2]   Done                    utm_medium=web3x
[3]   Done                    utm_name=web3xcss
ubuntu@instance-20240830-1406:~$ [debug] Command-line config: ['-vU', 'https://www.reddit.com/r/funnyvideos/comments/1f8xxze/three_hairy_guys_doing_some_practicals/?utm_source=share']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (zip)
[debug] Python 3.10.12 (CPython aarch64 64bit) - Linux-6.5.0-1024-oracle-aarch64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2
[debug] Optional libraries: certifi-2020.06.20, requests-2.25.1, secretstorage-3.3.1, sqlite3-3.37.2, urllib3-1.26.5
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[Reddit] Extracting URL: https://www.reddit.com/r/funnyvideos/comments/1f8xxze/three_hairy_guys_doing_some_practicals/?utm_source=share
[Reddit] 1f8xxze: Downloading JSON metadata
WARNING: [Reddit] 1f8xxze: Failed to parse JSON: Expecting value in '': line 1 column 1 (char 0)
[Reddit] www.reddit.com request failed, retrying with old.reddit.com
[Reddit] 1f8xxze: Downloading JSON metadata
ERROR: [Reddit] 1f8xxze: 1f8xxze: Failed to parse JSON (caused by JSONDecodeError("Expecting value in '': line 1 column 1 (char 0)")); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 740, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/reddit.py", line 261, in _real_extract
    data = self._download_json(
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1139, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1106, in download_handle
    return parse(self, content, video_id, transform_source=transform_source, fatal=fatal, errnote=errnote), urlh
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1094, in parse
    return getattr(ie, parser)(content, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1081, in _parse_json
    self.__print_error('Failed to parse JSON' if errnote is None else errnote, fatal, video_id, ve)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1064, in __print_error
    raise ExtractorError(f'{video_id}: {errnote}', cause=err)

  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 552, in decode
  File "/usr/lib/python3.10/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.10/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 1078, in _parse_json
    return json.loads(
  File "/usr/lib/python3.10/json/__init__.py", line 359, in loads
    return cls(**kw).decode(s)
  File "/usr/local/bin/yt-dlp/yt_dlp/utils/_utils.py", line 560, in decode
json.decoder.JSONDecodeError: Expecting value in '': line 1 column 1 (char 0)
```

### PR DESCRIPTION
**Pull request by pzhlkj6612**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This is a small and quick fix.

Fixes #10924.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Review by Grub4K — APPROVED** [review]

File: yt_dlp/extractor/reddit.py [hunk 1]
```diff
@@ -259,7 +259,9 @@ def _real_extract(self, url):
             fallback_host = 'old.reddit.com' if host != 'old.reddit.com' else 'www.reddit.com'
             self.to_screen(f'{host} request failed, retrying with {fallback_host}')
             data = self._download_json(
-                f'https://{fallback_host}/{slug}/.json', video_id, expected_status=403)
+                f'https://{fallback_host}/{slug}/.json', video_id, fatal=False, expected_status=403)
+            if not data:
+                self.raise_login_required('Account authentication is required')
```
[the remark this answers is not in the document]
Comment (pzhlkj6612) [reply yt_dlp/extractor/reddit.py:264]: OK, I'll change it to `try self._download_json(fatal=True) except json.decoder.JSONDecodeError`.

[the remark this answers is not in the document]
Comment (pzhlkj6612) [reply yt_dlp/extractor/reddit.py]: > It's specifically a JSONDecodeError that occurs when auth is needed, right? We'd still want to raise the HTTPError if we get a 404 etc, no?

Done. f81a5416d8497fd94ab7a95cb908810c36151fc7

**Review by bashonly — COMMENTED** [review]
Looks good, but I'm wondering if the `fallback_host` code is even useful at all anymore

**Comment by pzhlkj6612** [comment]
> I'm wondering if the `fallback_host` code is even useful at all anymore

Hmm, I agree. We should catch the exception at the first download. And I think we can always download that JSON from a fixed server (i.e., domain). Your opinion?

**Review by bashonly — APPROVED** [review]
Yep I think that's the way to go, and I went ahead and made the changes. Thanks for the PR

**Pull request merged**